client/providers/github_actions.go

```diff
@@ -68,25 +67,8 @@ func buildTokenURL(rawTokenURL, audience string) (string, error) {
 	return parsedURL.String(), nil
 }
 
-func (g *GithubOp) VerifyCICHash(ctx context.Context, idt []byte, expectedCICHash string) error {
-	cicHash, err := client.ExtractClaim(idt, "aud")
-	if err != nil {
-		return err
-	}
-
-	if cicHash != expectedCICHash {
-		return fmt.Errorf("aud claim doesn't match, got %q, expected %q", cicHash, expectedCICHash)
-	}
-
-	return nil
-}
-
-func (g *GithubOp) Issuer() string {
-	return githubIssuer
-}
-
-func (g *GithubOp) PublicKey(ctx context.Context, headers jws.Headers) (crypto.PublicKey, error) {
-	return client.DiscoverPublicKey(ctx, headers, githubIssuer)
+func (g *GithubOp) Verifier() verifier.ProviderVerifier {
+	return verifier.NewProviderVerifier(githubIssuer, "aud", verifier.ProviderVerifierOpts{})
```

Comment:
Github fail to verify a non-GQ PK Token as is the current behavior. Does this change that behavior?

```golang
func (*GithubOp) VerifyNonGQSig(context.Context, []byte, string) error {
	return client.ErrNonGQUnsupported
}
```
https://github.com/openpubkey/openpubkey/blob/main/client/providers/github_actions.go#L130C1-L132C2

Reply:
I see you have the GQOnly() check. I think you can just add that to the github ProviderVerifier